OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs

```diff
@@ -85,11 +84,28 @@ public void Tick(Actor self)
 				if (lastKnownActorIds.Contains(actor.Actor.ActorID))
 					continue;
 
+				var notificationPlayed = playedNotifications.Contains(actor.Trait.Info.Notification);
+
+				// Notify the actor that he's been discovered
+				foreach (var trait in actor.Actor.TraitsImplementing<INotifyDiscovered>())
+					trait.OnDiscovered(actor.Actor, self.Owner, !notificationPlayed);
+
+				var discoveredPlayer = actor.Actor.Owner;
+				if (!discoveredPlayers.Contains(discoveredPlayer))
+				{
+					// Notify the actor's owner that he's been discovered
+					foreach (var trait in discoveredPlayer.PlayerActor.TraitsImplementing<INotifyDiscovered>())
+						trait.OnDiscovered(actor.Actor, self.Owner, false);
+
+					discoveredPlayers.Add(discoveredPlayer);
```

Comment:
@abcdefg30: That's not what this does. This ensures that a player only notifies another player once that he's seen him.